### PR TITLE
feat(JOML): migrate rest of `LocationComponent`

### PR DESCRIPTION
**Pull request by skaldarnar**
## Module PRs

- [ ] https://github.com/Terasology/AdventureAssets/pull/35
- [ ] https://github.com/Terasology/Behaviors/pull/67
- [ ] https://github.com/Terasology/ClimateConditions/pull/45
- [ ] https://github.com/Terasology/DynamicCities/pull/87
- [ ] https://github.com/Terasology/GooKeeper/pull/42
- [ ] https://github.com/Terasology/Inventory/pull/32
- [ ] https://github.com/Terasology/ItemRendering/pull/17
- [ ] https://github.com/Terasology/LightAndShadow/pull/189

Contributes to #3832

### CODE REVIEW

File: engine/src/main/java/org/terasology/logic/location/LocationComponent.java [hunk 1]
```diff
@@ -59,10 +56,10 @@ public LocationComponent(Vector3fc position) {
 
     /**
      * @return local rotation of location component
-     *
-     * TODO: make this readonly Quaternionfc -- Michael Pollind
+     *         <p>
+     *         TODO: make this readonly Quaternionfc -- Michael Pollind
```
Comment (pollend) [inline engine/src/main/java/org/terasology/logic/location/LocationComponent.java:60]: you can delete this line

**Review by pollend — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/terasology/logic/location/LocationComponent.java [hunk 2]
```diff
@@ -193,11 +198,9 @@ public void setWorldPosition(Vector3fc pos) {
...
         LocationComponent parentLoc = parent.getComponent(LocationComponent.class);
         if (parentLoc != null) {
-            this.position.sub(JomlUtil.from(parentLoc.getWorldPosition(new org.joml.Vector3f())));
-            this.position.scale(1f / parentLoc.getWorldScale());
-            Quat4f rot = new Quat4f(0, 0, 0, 1);
-            rot.inverse(JomlUtil.from(parentLoc.getWorldRotation(new Quaternionf())));
-            rot.rotate(this.position, this.position);
+            this.position.sub(parentLoc.getWorldPosition(new Vector3f()));
+            this.position.div(parentLoc.getWorldScale());
+            this.position.rotate(parentLoc.getWorldRotation(new Quaternionf()).invert());
```
Comment (pollend) [inline engine/src/main/java/org/terasology/logic/location/LocationComponent.java:203]:
```suggestion
            this.position.rotate(parentLoc.getWorldRotation(new Quaternionf()).conjugate());
```

File: engine/src/main/java/org/terasology/logic/location/LocationComponent.java [hunk 3]
```diff
@@ -210,9 +213,8 @@ public void setWorldRotation(Quaternionfc value) {
         setLocalRotation(value);
         LocationComponent parentLoc = parent.getComponent(LocationComponent.class);
         if (parentLoc != null) {
-            Quat4f worldRot = JomlUtil.from(parentLoc.getWorldRotation(new Quaternionf()));
-            worldRot.inverse();
-            this.rotation.mul(worldRot, this.rotation);
+            Quaternionf worldRot = parentLoc.getWorldRotation(new Quaternionf()).invert();
```
Comment (pollend) [inline engine/src/main/java/org/terasology/logic/location/LocationComponent.java:216]:
```suggestion
            Quaternionf worldRot = parentLoc.getWorldRotation(new Quaternionf()).conjugate();
```

**Review by pollend — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/terasology/logic/ai/SimpleAISystem.java [hunk 4]
```diff
@@ -88,7 +89,7 @@ public void update(float delta) {
                 drive.set(targetDirection);
 
                 float yaw = (float) Math.atan2(targetDirection.x, targetDirection.z);
-                location.getLocalRotation().set(0, 1, 0, yaw);
+                location.setLocalRotation(new Quaternionf().setAngleAxis(yaw, 0, 1, 0));
```
Comment (pollend) [inline engine/src/main/java/org/terasology/logic/ai/SimpleAISystem.java:92]: I changed this. I'm pretty sure this is suppose to be axis angle.

**Review by pollend — APPROVED** [review]

File: engine/src/main/java/org/terasology/logic/ai/SimpleAISystem.java (hunk 4 above)
Comment (skaldarnar) [reply engine/src/main/java/org/terasology/logic/ai/SimpleAISystem.java:92]: After seeing this bug in GooKeeper I'm pretty sure, too :sweat_smile:

File: engine/src/main/java/org/terasology/logic/location/LocationComponent.java [hunk 5]
```diff
@@ -210,9 +213,8 @@ public void setWorldRotation(Quaternionfc value) {
         setLocalRotation(value);
         LocationComponent parentLoc = parent.getComponent(LocationComponent.class);
         if (parentLoc != null) {
-            Quat4f worldRot = JomlUtil.from(parentLoc.getWorldRotation(new Quaternionf()));
-            worldRot.inverse();
-            this.rotation.mul(worldRot, this.rotation);
+            Quaternionf worldRot = parentLoc.getWorldRotation(new Quaternionf()).conjugate();
+            this.rotation.mul(worldRot);
```
Comment (pollend) [inline engine/src/main/java/org/terasology/logic/location/LocationComponent.java:217]:
```suggestion
            this.rotation.premul(worldRot);
```

**Review by pollend — CHANGES_REQUESTED** [review]

**Review by pollend — APPROVED** [review]

**Pull request merged**